src/manifolds/Segre.jl

```diff
@@ -0,0 +1,473 @@
+        if !isnothing(e)
+            return e
+        end
+    end
+
+    return nothing
+end
+
+"""
+    function check_vector(M::Segre{ℝ, V}, p, v, kwargs...)
+
+Check whether `v` is a tangent vector to `p` on `M`, i.e. after `check_point(M, p)`, `v` has to be of same dimension as `p` and orthogonal to `p`. The tolerance can be set using the `kwargs...`.
+"""
+function check_vector(M::Segre{ℝ,V}, p, v, kwargs...) where {V}
+    e = check_point(M, p, kwargs...)
+    if !isnothing(e)
+        return e
+    end
+
+    for (x, xdot, n) in zip(p[2:end], v[2:end], V)
+        # check_vector(::AbstractSphere, ...) uses isapprox to compare the dot product to 0, which by default sets atol=0
+        e = check_vector(
+            Sphere(n - 1)::AbstractSphere,
+            x,
+            xdot;
+            rtol=1e-10,
+            atol=1e-10,
+            kwargs...,
+        )
+        if !isnothing(e)
+            return e
+        end
+    end
+
+    return nothing
+end
+
+"""
+    function get_coordinates(M::Segre{𝔽, V}, p, v; kwargs...)
+"""
+function get_coordinates(M::Segre{𝔽,V}, p, v; kwargs...) where {𝔽,V}
+    @assert(is_point(M, p))
+    @assert(is_vector(M, p, v))
+
+    coords = [
+        v[1],
+        [
+            get_coordinates(Sphere(n - 1), x, xdot, DefaultOrthonormalBasis(); kwargs...) for (n, x, xdot) in zip(V, p[2:end], v[2:end])
+        ]...,
+    ]
+
+    return vcat(coords...)
+end
+
+"""
+    function get_vector( M::Segre{𝔽, V}, p, X; kwargs...)
+"""
+function get_vector(M::Segre{𝔽,V}, p, X; kwargs...) where {𝔽,V}
+    @assert(is_point(M, p))
+
+    X_ = deepcopy(X)
+    v = eltype(p)[[] for _ in p] # Initialize
+    v[1] = [X_[1]]
+    X_ = X_[2:end]
+    for (i, n) in enumerate(V)
+        v[i + 1] = get_vector(
+            Sphere(n - 1),
+            p[i + 1],
+            X_[1:(n - 1)],
+            DefaultOrthonormalBasis();
+            kwargs...,
+        )
+        X_ = X_[n:end]
+    end
+
+    @assert(length(X_) == 0)
+    check_vector(M, p, v)
+
+    return v
+end
+
+@doc raw"""
+    function inner(M::Segre{ℝ, V}, p, u, v,)
+
+Inner product between two tangent vectors ``u = (\nu, u_1, \dots, u_d)`` and ``v = (\xi, v_1, \dots, v_d)`` at ``p \doteq (\lambda, x_1, \dots, x_d)``. This inner product is obtained by embedding the Segre manifold in the space of tensors equipped with the Euclidean metric:
+````math
+    \langle u, v \rangle_{p} = \nu \xi + \lambda^2 (\langle u_1, v_1 \rangle_{x_1} + \dots + \langle u_d, v_d \rangle_{x_d}),
+````
+where ``\nu``, ``\xi \in T_{\lambda} ℝ^{+} = ℝ`` and ``u_i``, ``v_i \in T_{x_i} S^{n_i - 1} \subset ℝ^{n_i}``.
+"""
+function inner(M::Segre{ℝ,V}, p, u, v) where {V}
+    return u[1][1] * v[1][1] + p[1][1]^2 * dot(u[2:end], v[2:end])
+end
+
+@doc raw"""
+    function norm(M::Segre{𝔽, V}, p, v)
+
+Norm of tangent vector ``v`` at ``p``.
+"""
+function norm(M::Segre{𝔽,V}, p, v) where {𝔽,V}
+    return sqrt(inner(M, p, v, v))
+end
```

Comment:
```suggestion
```
This is the default anyways, so it should not be necessary, cf https://github.com/JuliaManifolds/ManifoldsBase.jl/blob/76d6a5fb05a1593325dd342ae780dd89f55afbcd/src/ManifoldsBase.jl#L942-L948